kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/RemoveFile.java

```diff
@@ -36,8 +42,95 @@ public class RemoveFile {
           .add("deletionVector", DeletionVectorDescriptor.READ_SCHEMA, true /* nullable */)
           .add("baseRowId", LongType.LONG, true /* nullable */)
           .add("defaultRowCommitVersion", LongType.LONG, true /* nullable */);
-  // TODO: Currently, Kernel doesn't create RemoveFile actions internally, nor provides APIs for
-  //  connectors to generate and commit them. Once we have the need for this, we should ensure
-  //  that the baseRowId and defaultRowCommitVersion fields of RemoveFile actions are correctly
-  //  populated to match the corresponding AddFile actions.
+  // TODO: Currently Kernel doesn't support RemoveFile actions when rowTracking is enabled (or have
+  //   any public API for generating RemoveFile actions). Once we
+  //   do this we need to ensure that the baseRowId and defaultRowCommitVersion fields are correctly
+  //   populated to match the corresponding AddFile actions
+
+  /** Constructs an {@link RemoveFile} action from the given 'RemoveFile' {@link Row}. */
+  public RemoveFile(Row row) {
+    super(row);
+  }
+
+  public String getPath() {
+    return row.getString(getFieldIndex("path"));
+  }
+
+  public Optional<Long> getDeletionTimestamp() {
+    return row.isNullAt(getFieldIndex("deletionTimestamp"))
+        ? Optional.empty()
+        : Optional.of(row.getLong(getFieldIndex("deletionTimestamp")));
+  }
+
+  public boolean getDataChange() {
+    return row.getBoolean(getFieldIndex("dataChange"));
+  }
+
+  public Optional<Boolean> getExtendedFileMetadata() {
+    return row.isNullAt(getFieldIndex("extendedFileMetadata"))
```

Comment:
super nit: consider define some static constants for "extendedFileMetadata" as it appears multiple times.

Reply:
I don't think it's any more than every other field name in the remove schema?